src/utils.js

```diff
@@ -128,6 +128,26 @@ let sleep = (ms) => {
   return new Promise(resolve => setTimeout(resolve, ms));
 }
 
+const getXmlWidgetValue = (xml, widgetName, defaultValue = '') => {
+  try {
+    if (xml.widget && xml.widget[widgetName]) {
+      const value = xml.widget[widgetName];
+      if (Array.isArray(value)) {
+        if (typeof value[0] === 'object') {
+          return value[0]._;
+        } else if (typeof value[0] === 'string') {
+          return value[0];
+        }
+      } else {
+        return value;
+      }
+    }
+  } catch (error) {
+    // DO NOTHING
+  }
+  return defaultValue;
+};
+
```

Comment:
I'd like to suggest some revisions to enhance its functionality and maintainability. Here's why I believe the updated version would be beneficial:

- Handling of Undefined Cases: In the original code, if the first value in an array is neither an object nor a string, the function implicitly returns undefined. The revised code explicitly handles this case, making the function's behavior more predictable and understandable.

- Consistency in Return Types: The revised version ensures that the default return value is always a string, which aligns with the defaultValue parameter's expected type. This consistency prevents potential type-related issues.
(* Be aware that in JavaScript, null is also treated as a type of object, so caution is needed.)

```
// Function to retrieve the value of a specific widget from XML data
const getXmlWidgetValue = (xml, widgetName, defaultValue = '') => {
  // Helper function to safely get the first value from an array
  const safeGetFirstValue = (values) => {
    if (!Array.isArray(values) || values.length === 0) {
      return undefined;
    }

    const firstValue = values[0];
    // If the first element is an object and has the `_' property, return its value
    if (firstValue && typeof firstValue === 'object' && '_`' in firstValue) {
      return firstValue._;
    }

    return firstValue;
  };

  try {
    // Check if xml.widget is an object and contains the specified widget name
    if (xml.widget && typeof xml.widget === 'object' && widgetName in xml.widget) {
      const value = xml.widget[widgetName];
      const extractedValue = safeGetFirstValue(value);

      return extractedValue !== undefined ? extractedValue : value;
    }
  } catch (error) {
    console.error(`Error while getting widget value: ${error}`);
  }

  return typeof defaultValue === 'string' ? defaultValue : '';
};

```

Reply:
@takuyaW Thank you, watanabe-san. i have applied the feedback